### PR TITLE
feat(flutterfire_ui): Add German localization language support

### PR DESCRIPTION
**Pull request by mkoehne**
## Description

Added German localization support of flutterfire_ui.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [ ] All existing and new tests are passing.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by russellwheatley** [comment]
Hey @mkoehne, thanks for the PR! Do you mind getting a fellow German speaking developer to proofread & approve your translations (We all make mistakes 😄), please?

**Comment by devj3ns** [comment]
Hey @RedBrogdon @mkoehne, I am a german speaking developer and the translation look good to me :)

Only the `isNotАValidEmailErrorText` is missing a german translation.

File: packages/flutterfire_ui/lib/src/i10n/lang/de.dart [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+    this.verifyingPhoneNumberViewTitle = 'Code aus SMS eingeben',
+    this.unknownError = 'Ein unbekannter Fehler ist aufgetreten',
+    this.smsAutoresolutionFailedError =
+        'Der SMS-Code konnte nicht automatisch aufgelöst werden. Bitte geben Sie Ihren Code manuell ein',
+    this.smsCodeSentText = 'SMS-Code wurde verschickt',
+    this.sendingSMSCodeText = 'SMS-Code versenden...',
+    this.verifyingSMSCodeText = 'SMS-Code überprüfen...',
+    this.enterSMSCodeText = 'SMS-Code eingeben',
+    this.emailIsRequiredErrorText = 'E-Mail ist erforderlich',
+    this.isNotAValidEmailErrorText = 'Provide a valid email',
```
Comment (nilsreichardt) [inline packages/flutterfire_ui/lib/src/i10n/lang/de.dart:196]: Translation is missing

**Comment by mkoehne** [comment]
Hey @russellwheatley yes of course.
Thanks you @devj3ns for the review. I have added the translation for `isNotАValidEmailErrorText`.

**Review by russellwheatley — APPROVED** [review]
LGTM

**Review by nilsreichardt — APPROVED** [review]
LGTM

**Pull request merged**